src/lib/core/components/visualizations/implementations/MosaicVisualization.tsx

```diff
@@ -240,15 +248,22 @@ function MosaicViz(props: Props) {
         vizConfig.xAxisVariable,
         vizConfig.yAxisVariable
       );
-      const response = isTwoByTwo
-        ? dataClient.getTwoByTwo(
-            computation.type,
-            params as MosaicRequestParams
-          )
-        : dataClient.getMosaic(computation.type, params as MosaicRequestParams);
-      return isTwoByTwo
-        ? twoByTwoResponseToData(await response)
-        : mosaicResponseToData(await response);
+
+      if (isTwoByTwo) {
```

Comment:
Regarding your question
> The method I implemented of using either the MosaicData or TwoByTwoData type in the mosaic viz component--maybe better to use io-ts for this?

Is this where you mean?  I think it's fine to make two different requests and have data-api.ts validate them with two different validators...

Reply:
Yes, here and the above. Just seems like I'm doing the "either" thing io-ts is good at

Reply:
I agree with @bobular. This looks fine to me, as well.